src/server/pfs/server/obj_block_api_server.go

```diff
@@ -1109,8 +1113,8 @@ type blockWriter struct {
 	mu      sync.Mutex
 }
 
-func (s *objBlockAPIServer) newBlockWriter(block *pfsclient.Block) (*blockWriter, error) {
-	w, err := s.objClient.Writer(s.blockPath(block))
+func (s *objBlockAPIServer) newBlockWriter(ctx context.Context, block *pfsclient.Block) (*blockWriter, error) {
+	w, err := s.objClient.Writer(ctx, s.blockPath(block))
 	if err != nil {
 		return nil, err
 	}
```

Comment:
So it looks like all the changes in this file are moving around the ctx parameter. Are these actual semantic changes or are they refactors?

Reply:
They're somewhat semantic changes—I had to changed the `obj.Client` interface (in `src/server/pkg/obj/obj.go` to take a context in all of its calls, so that the various object clients (amazon_client, google_client, etc) could report spans to Jaeger for object store RPCs that were correctly associated with the surrounding Pachyderm RPCs. In other words, when you call `PutFile`, the `PutOject` calls associated with that show up underneath the original call in Jaeger because of the context propagation. But because of the interface change, I had to change all of the callers, which is all of these call sites are changed.

Incidentally, I got rid of the ctx that the Google client constructor took, because it was just using that context to make RPCs, and now it uses the context that it recieves for each call (which I'm pretty sure has better cancellation semantics anyway)